src/lib.rs

```diff
@@ -235,40 +170,36 @@ impl UPClientZenoh {
         }
     }
 
-    fn to_upayload_format(encoding: &Encoding) -> Option<UPayloadFormat> {
-        let Ok(value) = encoding.suffix().parse::<i32>() else {
-            return None;
-        };
-        UPayloadFormat::from_i32(value)
-    }
-
-    fn uattributes_to_attachment(uattributes: &UAttributes) -> anyhow::Result<AttachmentBuilder> {
-        let mut attachment = AttachmentBuilder::new();
-        attachment.insert("", &UATTRIBUTE_VERSION.to_le_bytes());
-        attachment.insert("", &uattributes.write_to_bytes()?);
-        Ok(attachment)
+    fn uattributes_to_attachment(uattributes: &UAttributes) -> anyhow::Result<ZBytes> {
+        let attachments = [
+            UATTRIBUTE_VERSION.to_le_bytes().to_vec(),
+            uattributes.write_to_bytes()?,
+        ];
+        Ok(attachments.iter().collect::<ZBytes>())
     }
 
-    fn attachment_to_uattributes(attachment: &Attachment) -> anyhow::Result<UAttributes> {
-        let mut attachment_iter = attachment.iter();
-        if let Some((_, value)) = attachment_iter.next() {
-            let version = *value.as_slice().first().ok_or_else(|| {
+    fn attachment_to_uattributes(attachment: &ZBytes) -> anyhow::Result<UAttributes> {
+        let mut attachment_iter = attachment.iter::<Vec<u8>>();
```

Comment:
You can use `Cow<[u8]>` to avoid unnecessary allocations when data is contiguous in memory.

```suggestion
        let mut attachment_iter = attachment.iter::<Cow<[u8]>>();
```